vignettes/RleTricks.Rmd

```diff
@@ -0,0 +1,57 @@
+---
+title: "Rle Tips and Tricks"
+author: "Patrick Aboyoun"
+date: "`r format(Sys.time(), '%d %B, %Y')`"
+package: S4Vectors
+vignette: >
+  %\VignetteIndexEntry{Vignette Title}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+output:
+    BiocStyle::html_document
+---
+
+    <<Rle-rollmean>>=
+    rollmeanRle <- function (x, k)
+    {
+      n <- length(x)
+      cumsum(c(Rle(sum(window(x, 1, k))), window(x, k + 1, n) - window(x, 1, n - k))) / k
+    }

```

Comment:
Use a code block. Since the code will not be evaluated, also add `eval=FALSE`. Remove the indent and name the code block the name in `<< ... >>=`. For example, the above should be

    ```{r Rle-rollmean, eval=FALSE}
    rollmeanRle <- function (x, k)
    {
        n <- length(x)
        cumsum(c(Rle(sum(window(x, 1, k))), window(x, k + 1, n) - window(x, 1, n - k))) / k
    }
    ```